python/pyspark/sql/tests/test_python_datasource.py

```diff
@@ -236,12 +236,10 @@ def reader(self, schema) -> "DataSourceReader":
 
         self.spark.dataSource.register(InMemoryDataSource)
         df = self.spark.read.format("memory").load()
-        self.assertEqual(df.rdd.getNumPartitions(), 3)
```

Comment:
nit, if we want to validate number of partitions in this case, we might use `spark_partition_id`

Reply:
Made a PR: https://github.com/apache/spark/pull/45720